iceoryx2-ffi/cxx/include/iox2/config.hpp

```diff
@@ -212,6 +212,18 @@ class Event {
     auto event_id_max_value() && -> size_t;
     /// Set the largest event id supported by the event service
     void set_event_id_max_value(size_t value) &&;
+    /// Defines the event id value that is emitted after a new notifier was created.
+    auto notifier_created_event() && -> iox::optional<size_t>;
+    /// Sets the event id value that is emitted after a new notifier was created.
+    void set_notifier_created_event(iox::optional<size_t> value) &&;
```

Comment:
This is intentional. In the builder it is no longer an optional but for the global config and the config file it is an optional. Some = emit event, None = no event.

Reply:
Wouldn't it make sense to have the same API with `disable`?

Reply:
No. In Rust the config is a struct where all members are public and can be set by the user directly. It is also serializable and stored in the `iceoryx2.toml` file. This config entry is still an `Option<EventId>` and it does not make sense to split it up into two fields since then we confuse the user and may have entries that are in contradiction to each other.

We require here a C++ API since we cannot access the struct members directly in C++. But the API shall reflect the underlying structure of the Rust structs/members. Therefore this method has still an optional.